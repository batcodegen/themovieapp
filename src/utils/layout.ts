import {I18nManager} from 'react-native';

const isRTL: boolean = I18nManager.isRTL;
const textinputAlignOnRTL: 'right' | 'left' = isRTL ? 'right' : 'left';
const currentLanguageLocale: string = isRTL ? 'ar-AE' : 'en-US';

export {isRTL, textinputAlignOnRTL, currentLanguageLocale};
