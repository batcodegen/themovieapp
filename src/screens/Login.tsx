import {
  View,
  Text,
  ImageBackground,
  Image,
  StyleSheet,
  Pressable,
  I18nManager,
} from 'react-native';
import React, {useEffect, useState} from 'react';
import {useTailwind} from 'tailwind-rn';
import {useTheme} from '@react-navigation/native';
import {BlurView} from '@react-native-community/blur';
import {useTranslation} from 'react-i18next';
import EmailInput from '../components/EmailInput';
import RNRestart from 'react-native-restart';
import PasswordInput from '../components/PasswordInput';
import Button from '../components/Button';
import {useSafeAreaInsets} from 'react-native-safe-area-context';
import {isRTL} from '../utils/layout';
import {useLogin} from '../hooks/useLogin';

const loginBackgroundImg = require('../assets/loginbg.png');
const Login = () => {
  const tw = useTailwind();
  const {colors} = useTheme();
  const insets = useSafeAreaInsets();
  const {t, i18n} = useTranslation();
  const [email, setEmail] = useState<string>('');
  const [password, setPassword] = useState<string>('');
  const [isValidEmail, setIsValidEmail] = useState<boolean>(false);
  const [isValidPassword, setIsValidPassword] = useState<boolean>(false);
  const {callLoginApi} = useLogin();

  useEffect(() => {
    if (isRTL) {
      i18n.changeLanguage('ar');
    } else {
      i18n.changeLanguage('en');
    }
  }, [isRTL]);

  const checkIsDisabled: boolean = !isValidEmail || !isValidPassword;

  const onAuthenticate = () => {
    callLoginApi(email, password);
  };

  return (
    <ImageBackground
      source={loginBackgroundImg}
      style={[tw('flex-1 bg-opacity-5'), {backgroundColor: colors.primary}]}>
      <BlurView
        style={tw('absolute top-0 bottom-0 right-0 left-0')}
        blurType="light"
        blurAmount={1}
        reducedTransparencyFallbackColor="white"
      />
      <Pressable
        style={[styles.langContainer, {top: insets.top + 10}]}
        onPress={() => {
          if (isRTL) {
            I18nManager.forceRTL(false);
          } else {
            I18nManager.forceRTL(true);
          }
          RNRestart.restart();
        }}>
        <Text style={styles.switch}>{isRTL ? 'EN' : 'AR'}</Text>
      </Pressable>
      <View style={styles.container}>
        <Image
          source={require('../assets/applogo.png')}
          style={styles.applogo}
        />
        <Text style={[tw('font-bold text-lg text-center'), styles.title]}>
          THE MOVIE APP
        </Text>
        <EmailInput
          onSubmitText={(text, isValid) => {
            setEmail(text);
            setIsValidEmail(isValid);
          }}
          value={email}
        />
        <PasswordInput
          value={password}
          onSubmitText={(text, isValid) => {
            setPassword(text);
            setIsValidPassword(isValid);
          }}
        />
        <Button
          title={t('login.loginBtn')}
          disabled={checkIsDisabled}
          onButtonPress={onAuthenticate}
        />
      </View>
    </ImageBackground>
  );
};

export default Login;

const styles = StyleSheet.create({
  container: {
    justifyContent: 'center',
    flex: 1,
    alignItems: 'center',
  },
  applogo: {width: 100, height: 100},
  title: {color: '#f0af0cfd'},
  langContainer: {
    position: 'absolute',
    right: 15,
    zIndex: 1,
  },
  switch: {fontWeight: 'bold', fontSize: 20, color: 'black'},
});
