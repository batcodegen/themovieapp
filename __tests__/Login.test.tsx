import React from 'react';
import {screen} from '@testing-library/react-native';
import Login from '../src/screens/Login';
import {renderWithProviders} from './utils/test-utils';

describe('Login', () => {
  test('renders Login correctly', async () => {
    renderWithProviders(<Login />);
    expect(screen.toJSON()).toMatchSnapshot();
  });
});
