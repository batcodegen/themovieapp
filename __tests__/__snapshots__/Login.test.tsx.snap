// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login renders Login correctly 1`] = `
<View
  accessibilityIgnoresInvertColors={true}
  style={
    [
      "flex-1 bg-opacity-5",
      {
        "backgroundColor": "#000",
      },
    ]
  }
>
  <Image
    source={
      {
        "testUri": "../../../src/assets/loginbg.png",
      }
    }
    style={
      [
        {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        {
          "height": undefined,
          "width": undefined,
        },
        undefined,
      ]
    }
  />
  <BlurView
    blurAmount={1}
    blurType="light"
    reducedTransparencyFallbackColor="white"
    style={
      [
        {
          "backgroundColor": "transparent",
        },
        "absolute top-0 bottom-0 right-0 left-0",
      ]
    }
  />
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      [
        {
          "position": "absolute",
          "right": 15,
          "zIndex": 1,
        },
        {
          "top": 10,
        },
      ]
    }
  >
    <Text
      style={
        {
          "color": "black",
          "fontSize": 20,
          "fontWeight": "bold",
        }
      }
    >
      AR
    </Text>
  </View>
  <View
    style={
      {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <Image
      source={
        {
          "testUri": "../../../src/assets/applogo.png",
        }
      }
      style={
        {
          "height": 100,
          "width": 100,
        }
      }
    />
    <Text
      style={
        [
          "font-bold text-lg text-center",
          {
            "color": "#f0af0cfd",
          },
        ]
      }
    >
      THE MOVIE APP
    </Text>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        [
          {
            "alignItems": "center",
            "borderRadius": 10,
            "borderWidth": 1,
            "flexDirection": "row",
            "marginBottom": 10,
            "marginTop": 20,
            "paddingHorizontal": 10,
            "width": "70%",
          },
          {
            "borderColor": undefined,
          },
        ]
      }
      testID="email-open-button"
    >
      <Text
        allowFontScaling={false}
        selectable={false}
        style={
          [
            {
              "color": "white",
              "fontSize": 18,
            },
            undefined,
            {
              "fontFamily": "Fontisto",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            {},
          ]
        }
      >
        
      </Text>
      <TextInput
        editable={false}
        placeholder="login.email"
        placeholderTextColor="grey"
        pointerEvents="none"
        style={
          {
            "color": "white",
            "flex": 1,
            "overflow": "hidden",
            "padding": 10,
            "textAlign": "left",
          }
        }
        value=""
      />
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        [
          {
            "alignItems": "center",
            "borderRadius": 10,
            "borderWidth": 1,
            "flexDirection": "row",
            "marginBottom": 10,
            "marginTop": 20,
            "paddingHorizontal": 10,
            "width": "70%",
            "zIndex": 2,
          },
          {
            "borderColor": undefined,
          },
        ]
      }
      testID="password-open-button"
    >
      <Text
        allowFontScaling={false}
        selectable={false}
        style={
          [
            {
              "color": "white",
              "fontSize": 18,
            },
            undefined,
            {
              "fontFamily": "Feather",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            {},
          ]
        }
      >
        
      </Text>
      <TextInput
        editable={false}
        placeholder="login.password"
        placeholderTextColor="grey"
        pointerEvents="none"
        secureTextEntry={true}
        style={
          {
            "color": "white",
            "flex": 1,
            "overflow": "hidden",
            "padding": 10,
            "textAlign": "left",
          }
        }
        value=""
      />
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": true,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#868583",
          "borderRadius": 10,
          "margin": 20,
          "opacity": 1,
          "padding": 10,
          "width": "70%",
        }
      }
    >
      <Text
        style={
          {
            "color": "white",
            "fontSize": 16,
            "fontWeight": "bold",
          }
        }
      >
        login.loginBtn
      </Text>
    </View>
  </View>
</View>
`;
